lib/invoker.js

```diff
@@ -24,21 +25,63 @@ const invoke = (userFunction) => {
     };
 };
 
-module.exports = (userFunctionUri, port) => {
-    const userFunction = lookUpFunction(userFunctionUri);
+const registerShutdown = (destroyHook, timeout, server, shutdownSignal) => {
+    return shutdownSignal.then(() => {
+        logger('Graceful shutdown started');
+        return new Promise(((resolve, reject) => {
+            server.tryShutdown(() => {
+                executeDestroy(destroyHook, timeout)
+                    .then(() => {
+                        logger('Graceful shutdown completed');
+                        resolve();
+                    }, (err) => {
+                        logger(`Error during $destroy hook execution: ${err.toString()}`);
+                        reject(err);
+                    });
+            });
+        }));
+    });
+};
+
+const executeInit = async (initHook, initTimeout) => {
+    logger('Calling $init hook (if any)');
+    return await guardWithTimeout(initHook, initTimeout)
```

Comment:
no need to await before returning in an async function

```suggestion
    return guardWithTimeout(initHook, initTimeout)
```